### PR TITLE
Remove OA flow rate dependency on Branch Maximum Flow Rate and deprecate the field

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
Addresses #5119 and makes other improvements. The main impact on users is that the Branch object field "Branch Maximum Flow Rate" is now deprecated.  A follow-up issue #5600 has been posted to actually remove the field from the object.
1. AirLoopFlow.DesSupply (design supply air mass flow rate for loop [kg/s]) is now based on AirLoopHVAC Design Supply Air Flow Rate instead of Branch Maximum Flow Rate.
2. The error check for Controller:OutdoorAir Minimum and Maximum Outdoor Air Flow Rate > the air loop max flow rate has been improved and moved to a better location.
3.  For air loops, replace all uses of Branch "Maximum Flow Rate" value with the AirLoopHVAC "Design Supply Air Flow Rate". (This extends item 1 above - when/if this all works - combine this note with note 1).
4.  Remove unitary system and furnace checks on flow vs branch flow rate.
   _Replace with checks against air loop design flow rate?_
5.  Remove lingering vestiges of branch flow rate in plant.
6. Remove several obsolete functions.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
  - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
  - NewFeature: This pull request includes code to add a new feature to EnergyPlus
  - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
  - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog
### Review Checklist

This will not be exhaustively relevant to every PR.
- [ ] Code style (parentheses padding, variable names)
- [ ] Functional code review (it has to work!)
- [ ] If defect, results of running current develop vs this branch should exhibit the fix
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- C++ checks:
  - [ ] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] ExpandObjects changes??
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by mjwitte** [comment]
The four "failed" test files are ones where airflow rates were fixed to be consistent with the same systems.  the one "warning" file gets rid of an warning message (again by fixing mismatched flow rates).

**Comment by mjwitte** [comment]
No new results differences introduced by eliminating use of the Branch max flow rate.

**Comment by mjwitte** [comment]
@rraustad I am attempting to remove any dependencies on Branch Max Flow Rate, and I ran into sizing blocks like [this](https://github.com/NREL/EnergyPlus/blob/c87e61b44bce627bc0fac6f0574d9b018015a3be/src/EnergyPlus/HVACUnitarySystem.cc#L2457-L2475) which are using branch flow rate and/or branch fan flow rate to set unitary system or coil flow rate.  Some (like this one) are prefaced with a comment questioning the need for this block.  Can you help me understand the intent of these, i.e. when they are needed?
Other similar blocks:
[Here](https://github.com/NREL/EnergyPlus/blob/c87e61b44bce627bc0fac6f0574d9b018015a3be/src/EnergyPlus/HVACUnitarySystem.cc#L2620-L2634)
[Here2](https://github.com/NREL/EnergyPlus/blob/c87e61b44bce627bc0fac6f0574d9b018015a3be/src/EnergyPlus/HVACUnitarySystem.cc#L1423-L1451)
Could these use the AirLoopHVAC design flow rate instead of branch flow rate?

**Comment by rraustad** [comment]
The first one (this, 2457-2475) was an attempt to size air flow when the 
parent could not. I can't recall when this was needed, maybe with a 
water coil since it does not have an air flow input. But that does not 
make sense now since the parent always sizes the flow rate. I would say 
you can probably comment this out with an added comment that it may be 
needed in the future to size system air flow and we are looking for when 
that is required.

The second one (Here, 2620-2634) was needed for when the branch was 
hardsized and system was autosized.
And the third (Here2, 1423-1451) sets up the speed ratios used for fan 
power control when using the fan curves in the Fan:OnOff object

On 4/1/2016 1:21 PM, Michael J. Witte wrote:

> @rraustad https://github.com/rraustad I am attempting to remove any 
> dependencies on Branch Max Flow Rate, and I ran into sizing blocks 
> like this 
> https://github.com/NREL/EnergyPlus/blob/c87e61b44bce627bc0fac6f0574d9b018015a3be/src/EnergyPlus/HVACUnitarySystem.cc#L2457-L2475 
> which are using branch flow rate and/or branch fan flow rate to set 
> unitary system or coil flow rate. Some (like this one) are prefaced 
> with a comment questioning the need for this block. Can you help me 
> understand the intent of these, i.e. when they are needed?
> Other similar blocks:
> Here 
> https://github.com/NREL/EnergyPlus/blob/c87e61b44bce627bc0fac6f0574d9b018015a3be/src/EnergyPlus/HVACUnitarySystem.cc#L2620-L2634
> Here2 
> https://github.com/NREL/EnergyPlus/blob/c87e61b44bce627bc0fac6f0574d9b018015a3be/src/EnergyPlus/HVACUnitarySystem.cc#L1423-L1451
> Could these use the AirLoopHVAC design flow rate instead of branch 
> flow rate?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> https://github.com/NREL/EnergyPlus/pull/5582#issuecomment-204477599

## 

Richard Raustad
Florida Solar Energy Center
1679 Clearlake Road
Cocoa, FL 32922
Ph: (321)638-1454
http://www.fsec.ucf.edu/en/

Program Director
Electric Vehicle Transportation Center
http://evtc.fsec.ucf.edu/

**Comment by mjwitte** [comment]
@nrel-bot-2 Are you out there?  Haven't heard from you since yesterday?

**Comment by nrel-bot-3** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

File: testfiles/1ZoneParameterAspect.idf [hunk 1]
```diff
@@ -427,7 +427,7 @@
 
   Branch,
     Air Loop Main Branch,    !- Name
-    2.0,                     !- Maximum Flow Rate {m3/s}
+    3.0,                     !- Maximum Flow Rate {m3/s}
```
Comment (rraustad) [inline testfiles/1ZoneParameterAspect.idf:430]: ??? Didn't you get rid of this input? Line 1400 BranchInputManager above.

Comment (mjwitte) [inline testfiles/1ZoneParameterAspect.idf:430]: True.  This shouldn't matter now.

**Comment by Myoldmopar** [comment]
So much cleanup going on here...a beautiful thing.  @rraustad have you reviewed all this?  I see you made the one comment above, and just wondering how much time I should spend reviewing/testing this.

**Comment by rraustad** [comment]
I have looked through all and it looks good. A few trst files should show if it passes muster.

Sent from my Sprint Samsung Galaxy S7.
-------- Original message --------From: Edwin Lee notifications@github.com Date: 4/28/16  1:01 PM  (GMT-05:00) To: NREL/EnergyPlus EnergyPlus@noreply.github.com Cc: Richard Raustad rraustad@fsec.ucf.edu, Mention mention@noreply.github.com Subject: Re: [NREL/EnergyPlus] Remove OA flow rate dependency on Branch Maximum
  Flow Rate and deprecate the field (#5582) 
So much cleanup going on here...a beautiful thing.  @rraustad have you reviewed all this?  I see you made the one comment above, and just wondering how much time I should spend reviewing/testing this.

**Comment by Myoldmopar** [comment]
Yeah test results and everything look good.  I'm inclined to merge.  I may build this locally to run additional tests unless you have some results to show off from your own testing already done @mjwitte ...

**Comment by Myoldmopar** [comment]
Ran some files locally and spot checked and things look awesome.  I thought you were really on to something when the runtime was cut in half with this branch during some tests 👍 ......but nope-fluke 👎.  Happily merging in this cleanliness.

**Pull request merged**